tests/pyccel/test_pyccel.py

```diff
@@ -110,7 +107,9 @@ def get_value(string, regex, conversion):
 def compare_pyth_fort_output_by_type( p_output, f_output, dtype=float ):
 
     if dtype is str:
-        assert(p_output.strip()==f_output.strip())
+        p_list = p_output.split()
+        f_list = f_output.split()
```

Comment:
Is the splitting still needed even with the concatenations used above?

Reply:
yes, Fortran adds spaces at the beginning of the printed line so without split string will not match.

Reply:
The original command used `strip()` which removes leading and trailing whitespace. This should therefore deal with spaces added at the beginning of the line while still letting you test the whitespace within the line

Reply:
compare string result is tricky. not like numbers.
actually code should be something similar to that.
```python
if dtype is str:
        p_list = p_output.split("\n")
        f_list = f_output.split("\n")
        # get firsl lines to compare
        p = p_list[0]
        f = f_list[0]
        # return the rest of the output to compare in next compare type
        p_output = "\n".join(p_list[1:] if len(p_list) > 1 else "")
        f_output = "\n".join(f_list[1:] if len(f_list) > 1 else "")
        assert(p.strip() == f.strip())
```
this can be easy to work with in case of printing one line of string.
and also can be a complex thing if you have a string with newlines inside, you should know the numbers of lines in the resulting output.
but what about this case.
```python
       print("hello world", 2020)
       #output in Fortran
       "hello world                  2020"
```
here for correct output, we should split with space. but if we do that then we should now how many words in the string.
so splitting this by whitespace and do one list comparison is better.

Reply:
Ok. I thought the output was already split at `\n` but it seems not so then you are right that `split()` is needed